### PR TITLE
Show steps for drag and drop designer

### LINKED ISSUE #3468: [feature][needs-docs] Add bg color option to form containers
Original commit: https://github.com/qgis/QGIS/commit/8d3f73a62a087d4842511563b15307ba4101e983 by elpaso

Because gray is boring.

### PR DESCRIPTION
**Pull request by DelazJ**
Fix #3468 (but this is mainly a move to steps-like description)

### CODE REVIEW

File: source/docs/user_manual/working_with_vector/vector_properties.rst [hunk 1]
```diff
@@ -1956,48 +1954,61 @@ shown for example in figure_fields_form_.
...
-* show the category as a group box (only available for tabs).
-
-With a double-click on a field label, you can also specify whether the label of
-its widget should be visible or not in the form.
-
-In case the layer is involved in ``one or many to many relations`` (see :ref:`vector_relations`),
-referencing layers are listed in the :guilabel:`Relations` frame and their form
-can be embedded in the current layer form by drag-and-drop. Like the other
-items, double-click the relation label to configure some options:
+#. Choose ``Drag and drop designer`` from the :guilabel:`Attribute editor
```
Comment (havatv) [inline source/docs/user_manual/working_with_vector/vector_properties.rst]:
```suggestion
#. Choose ``Drag and drop designer`` from the :guilabel:`Select attribute layout
```

File: source/docs/user_manual/working_with_vector/vector_properties.rst [hunk 2]
```diff
@@ -1956,48 +1954,61 @@ shown for example in figure_fields_form_.
...
+   layout` combobox. This enables the :guilabel:`Form Layout` panel next to the
```
Comment (havatv) [inline source/docs/user_manual/working_with_vector/vector_properties.rst]:
```suggestion
   editor` combobox. This enables the :guilabel:`Form Layout` panel next to the
```

File: source/docs/user_manual/working_with_vector/vector_properties.rst [hunk 3]
```diff
@@ -1956,48 +1954,61 @@ shown for example in figure_fields_form_.
...
+   :guilabel:`Available widgets` one, filled with existing fields.
```
Comment (havatv) [inline source/docs/user_manual/working_with_vector/vector_properties.rst]:
```suggestion
   :guilabel:`Available widgets` panel, filled with existing fields. 
```

File: source/docs/user_manual/working_with_vector/vector_properties.rst [hunk 4]
```diff
@@ -1956,48 +1954,61 @@ shown for example in figure_fields_form_.
...
-In case the layer is involved in ``one or many to many relations`` (see :ref:`vector_relations`),
-referencing layers are listed in the :guilabel:`Relations` frame and their form
-can be embedded in the current layer form by drag-and-drop. Like the other
-items, double-click the relation label to configure some options:
+#. Choose ``Drag and drop designer`` from the :guilabel:`Attribute editor
+   layout` combobox. This enables the :guilabel:`Form Layout` panel next to the
+   :guilabel:`Available widgets` one, filled with existing fields. 
+#. Select any fields you would not want to use in your :guilabel:`Form Layout`
+   panel and hit the |signMinus| button to remove them. Drag and drop fields
+   from the other panel to readd them. The same field can be added multiple times.
```
Comment (havatv) [inline source/docs/user_manual/working_with_vector/vector_properties.rst]:
```suggestion
   from the other panel to re-add them). The same field can be added multiple times.
```

File: source/docs/user_manual/working_with_vector/vector_properties.rst [hunk 5]
```diff
@@ -1956,48 +1954,61 @@ shown for example in figure_fields_form_.
...
-
-In case the layer is involved in ``one or many to many relations`` (see :ref:`vector_relations`),
-referencing layers are listed in the :guilabel:`Relations` frame and their form
-can be embedded in the current layer form by drag-and-drop. Like the other
-items, double-click the relation label to configure some options:
+#. Choose ``Drag and drop designer`` from the :guilabel:`Attribute editor
+   layout` combobox. This enables the :guilabel:`Form Layout` panel next to the
+   :guilabel:`Available widgets` one, filled with existing fields. 
+#. Select any fields you would not want to use in your :guilabel:`Form Layout`
+   panel and hit the |signMinus| button to remove them. Drag and drop fields
```
Comment (havatv) [inline source/docs/user_manual/working_with_vector/vector_properties.rst]:
```suggestion
   panel and hit the |signMinus| button to remove them (drag and drop fields
```

File: source/docs/user_manual/working_with_vector/vector_properties.rst [hunk 6]
```diff
@@ -1956,48 +1954,61 @@ shown for example in figure_fields_form_.
...
+   from the other panel to readd them. The same field can be added multiple times.
+#. Drag and drop fields within the :guilabel:`Form Layout` to reorder
+   their position.
+#. Add containers (tab or group frames) to associate fields that belong to
+   the same category and better structure the form.
+  
+   #. The first step is to use the |signPlus| icon to create a tab in which
+      fields and groups will be displayed
+   #. You can create as many containers as you want. If relevant, press again
```
Comment (havatv) [inline source/docs/user_manual/working_with_vector/vector_properties.rst]:
```suggestion
   #. You can create as many containers as you want. If relevant, press
```

File: source/docs/user_manual/working_with_vector/vector_properties.rst [hunk 7]
```diff
@@ -1956,48 +1954,61 @@ shown for example in figure_fields_form_.
...
+      the |signPlus| icon to create another tab or a group frame under
```
Comment (havatv) [inline source/docs/user_manual/working_with_vector/vector_properties.rst]:
```suggestion
      the |signPlus| icon again to create another tab or a group frame under
```

File: source/docs/user_manual/working_with_vector/vector_properties.rst [hunk 8]
```diff
@@ -1956,48 +1954,61 @@ shown for example in figure_fields_form_.
...
+      an existing tab (see figure_fields_layout_).
+
+      .. _figure_fields_layout:
+
+      .. figure:: img/attribute_editor_layout.png
+         :align: center
+
+         Dialog to create containers with the **Attribute editor layout**
+
+#. The next step is to assign to each container the relevant fields,
```
Comment (havatv) [inline source/docs/user_manual/working_with_vector/vector_properties.rst]:
```suggestion
#. The next step is to assign the relevant fields to each container,
```

File: source/docs/user_manual/working_with_vector/vector_properties.rst [hunk 9]
```diff
@@ -1956,48 +1954,61 @@ shown for example in figure_fields_form_.
...
+   by simple drag and drop. Groups and tabs can also be moved, the same way.
```
Comment (havatv) [inline source/docs/user_manual/working_with_vector/vector_properties.rst]:
```suggestion
   by simple drag and drop. Groups and tabs can also be moved in the same way.
```

File: source/docs/user_manual/working_with_vector/vector_properties.rst [hunk 10]
```diff
@@ -1956,48 +1954,61 @@ shown for example in figure_fields_form_.
...
+#. At their creation time you can set some properties of the containers.
```
Comment (havatv) [inline source/docs/user_manual/working_with_vector/vector_properties.rst]:
```suggestion
#. At their creation time you can set properties of the containers
```

File: source/docs/user_manual/working_with_vector/vector_properties.rst [hunk 11]
```diff
@@ -1956,48 +1954,61 @@ shown for example in figure_fields_form_.
...
+   At any time you can update them with a double-click. QGIS opens a form in
```
Comment (havatv) [inline source/docs/user_manual/working_with_vector/vector_properties.rst]:
```suggestion
   (they can updated them with a double-click). QGIS opens a form in
```

File: source/docs/user_manual/working_with_vector/vector_properties.rst [hunk 12]
```diff
@@ -1956,48 +1954,61 @@ shown for example in figure_fields_form_.
...
+   which you can:
+
+   * choose to hide or show the item label
+   * rename the container
+   * set over how many columns the embedded fields should be distributed
+   * enter an expression to control the container's visibility.
+     The expression will be re-evaluated everytime values in the form change
```
Comment (havatv) [inline source/docs/user_manual/working_with_vector/vector_properties.rst]:
```suggestion
     The expression will be re-evaluated every time values in the form change,
```

File: source/docs/user_manual/working_with_vector/vector_properties.rst [hunk 13]
```diff
@@ -1956,48 +1954,61 @@ shown for example in figure_fields_form_.
...
+     and the tab or group box shown/hidden accordingly
+   * show the container as a group box (only available for tabs).
+   * add a background color
+
+#. Likewise double-click a field label and you can specify whether the label
```
Comment (havatv) [inline source/docs/user_manual/working_with_vector/vector_properties.rst]:
```suggestion
#. Likewise, double-click a field label and you can specify whether the label
```

File: source/docs/user_manual/working_with_vector/vector_properties.rst [hunk 14]
```diff
@@ -1956,48 +1954,61 @@ shown for example in figure_fields_form_.
...
+   of its widget should be visible or not in the form.
+
+#. In case the layer is involved in ``one or many to many relations`` (see 
+   :ref:`vector_relations`), referencing layers are listed in the
+   :guilabel:`Relations` frame and their form can be embedded in the current
+   layer form by drag-and-drop. Like the other items, double-click the relation
```
Comment (havatv) [inline source/docs/user_manual/working_with_vector/vector_properties.rst]:
```suggestion
   layer form by drag-and-drop. As for the other items, double-click the relation
```

File: source/docs/user_manual/working_with_vector/vector_properties.rst [hunk 15]
```diff
@@ -1956,48 +1954,61 @@ shown for example in figure_fields_form_.
...
+   label to configure some options:
+
+   * choose to hide or show the item label
+   * show the link button
+   * show the unlink button
+
+#. Apply the layer's properties dialog
+#. Use the |identify| tool on a feature and it should display the new form.
```
Comment (havatv) [inline source/docs/user_manual/working_with_vector/vector_properties.rst]:
```suggestion
#. Use the |identify| tool on a feature, and it should display the new form.
```

File: source/docs/user_manual/working_with_vector/vector_properties.rst (hunk 5 above)
Comment (DelazJ) [reply source/docs/user_manual/working_with_vector/vector_properties.rst]: Not convinced parenthesis is appropriate here.

File: source/docs/user_manual/working_with_vector/vector_properties.rst (hunk 11 above)
Comment (DelazJ) [reply source/docs/user_manual/working_with_vector/vector_properties.rst]: Actually, the point is that the update dialog provides more options that the creation one. Putting that action into brackets minimizes it to me although it's what is described below

Comment (havatv) [reply source/docs/user_manual/working_with_vector/vector_properties.rst]: Good point.  With more options for the update, reformulation is necessary.  I have given it a try.

File: source/docs/user_manual/working_with_vector/vector_properties.rst (hunk 5 above)
Comment (havatv) [reply source/docs/user_manual/working_with_vector/vector_properties.rst]: We describe how to remove fields here.  Then I think it is appropriate to describe how to get them back in parenthesis (reversing the action).

**Pull request merged**